app/src/pages/playground/PlaygroundChatTemplate.tsx

```diff
@@ -290,7 +292,32 @@ function MessageEditor({
       </View>
     );
   }
+  if (messageMode === "json") {
+    return (
+      <View
+        paddingStart="size-200"
+        paddingEnd="size-200"
+        paddingTop="size-200"
+        paddingBottom="size-200"
+      >
+        <ChatMessageJSONContentEditor
+          playgroundInstanceId={playgroundInstanceId}
+          templateMessages={template.messages}
+          messageId={message.id}
+        />
+      </View>
+    );
+  }
   if (message.role === "tool") {
+    const toolMessageContent =
+      // transform empty values into empty object string
+      message.content == null || message.content === ""
+        ? "{}"
+        : // transform non-string values into a pretty-printed JSON string
+          typeof message.content === "string"
+          ? // assume strings are already JSON strings
+            message.content
+          : JSON.stringify(message.content, null, 2);
```

Comment:
can we remove the double ternary and pull this out into a function or something getInitialToolMessageContent or something